src/prognostic_equations/edmfx_entr_detr.jl

```diff
@@ -0,0 +1,281 @@
+#####
+##### EDMF entrainment detrainment
+#####
+
+"""
+   Return entrainment rate [1/s].
+
+   Inputs (everything defined on cell centers):
+   - params set with model parameters
+   - ᶜz, z_sfc, ᶜp, ᶜρ, - grid-scale height, surface height, grid-scale pressure and density
+   - buoy_flux_surface - buoyancy flux at the surface
+   - ᶜaʲ, ᶜwʲ, ᶜRHʲ, ᶜbuoyʲ - updraft area, physical vertical velocity,
+                                   relative humidity and buoyancy
+   - ᶜw⁰, ᶜRH⁰, ᶜbuoy⁰ - environment physical vertical velocity,
+                              relative humidity and buoyancy
+   - dt - timestep
+"""
+
+function pi_groups_entr(
+    params,
+    ᶜz::FT,
+    z_sfc::FT,
+    ᶜp::FT,
+    ᶜρ::FT,
+    buoy_flux_surface::FT,
+    ᶜaʲ::FT,
+    ᶜwʲ::FT,
+    ᶜRHʲ::FT,
+    ᶜbuoyʲ::FT,
+    ᶜw⁰::FT,
+    ᶜRH⁰::FT,
+    ᶜbuoy⁰::FT,
+    dt::FT,
+    ::NoEntrainment,
+) where {FT}
+    return FT(0)
+end
+
+function pi_groups_entr(
+    params,
+    ᶜz::FT,
+    z_sfc::FT,
+    ᶜp::FT,
+    ᶜρ::FT,
+    buoy_flux_surface::FT,
+    ᶜaʲ::FT,
+    ᶜwʲ::FT,
+    ᶜRHʲ::FT,
+    ᶜbuoyʲ::FT,
+    ᶜw⁰::FT,
+    ᶜRH⁰::FT,
+    ᶜbuoy⁰::FT,
+    dt::FT,
+    ::PiGroupsEntrainment,
+) where {FT}
+    if ᶜaʲ <= FT(0)
+        return FT(0)
+    else
+        g = CAP.grav(params)
+        turbconv_params = CAP.turbconv_params(params)
+        ᶜaʲ_max = TCP.max_area(turbconv_params)
+
+        # pressure scale height (height where pressure drops by 1/e)
+        ref_H = ᶜp / (ᶜρ * g)
+        # convective velocity
+        w_star = get_wstar(buoy_flux_surface)
+
+        # non-dimensional pi-groups
+        # TODO - using Π₁ blows things up
+        Π₁ =
+            (ᶜz - z_sfc) * (ᶜbuoyʲ - ᶜbuoy⁰) /
+            ((ᶜwʲ - ᶜw⁰)^2 + w_star^2 + eps(FT))
+        Π₃ = sqrt(ᶜaʲ)
+        Π₄ = ᶜRHʲ - ᶜRH⁰
+        Π₆ = (ᶜz - z_sfc) / ref_H
+        entr = max(
+            0,
+            ᶜwʲ / (ᶜz - z_sfc) * (
+                -4.013288 - 0.000968 * Π₁ + 0.356974 * Π₃ - 0.403124 * Π₄ +
+                1.503261 * Π₆
+            ),
+        )
+
+        entr = max(0, min(entr, 1 / dt))
+        return entr
+    end
+end
+
+function pi_groups_entr(
+    params,
+    ᶜz::FT,
+    z_sfc::FT,
+    ᶜp::FT,
+    ᶜρ::FT,
+    buoy_flux_surface::FT,
+    ᶜaʲ::FT,
+    ᶜwʲ::FT,
+    ᶜRHʲ::FT,
+    ᶜbuoyʲ::FT,
+    ᶜw⁰::FT,
+    ᶜRH⁰::FT,
+    ᶜbuoy⁰::FT,
+    dt::FT,
+    ::ConstantCoefficientEntrainment,
+) where {FT}
+    if ᶜaʲ <= FT(0)
```

Comment:
Do we want to keep this (and for other entrainment and detrainment closures except the pi group)?

Reply:
Probably not. It was only there because the pi-groups won't work for a<0. Last week we said that it makes sense for that case to have detr > entr. Does that still make sense this week?